src/video/wayland/SDL_waylandwindow.c

```diff
@@ -1894,42 +2126,108 @@ int Wayland_CreateWindow(_THIS, SDL_Window *window)
     wl_surface_set_opaque_region(data->surface, region);
     wl_region_destroy(region);
 
-    if (c->relative_mouse_mode) {
-        Wayland_input_lock_pointer(c->input);
-    }
-
-    /* Moved this call to ShowWindow: wl_surface_commit(data->surface); */
-    WAYLAND_wl_display_flush(c->display);
-
-    /* We may need to create an idle inhibitor for this new window */
-    Wayland_SuspendScreenSaver(_this);
-
-    #define IS_POPUP(window) \
+#define IS_POPUP(window) \
         (window->flags & (SDL_WINDOW_TOOLTIP | SDL_WINDOW_POPUP_MENU))
 #ifdef HAVE_LIBDECOR_H
     if (c->shell.libdecor && !IS_POPUP(window)) {
         data->shell_surface_type = WAYLAND_SURFACE_LIBDECOR;
     } else
 #endif
-    if (c->shell.xdg) {
+            if (c->shell.xdg) {
```

Comment:
Super minor spacing thing:

```suggestion
    if (c->shell.xdg) {
```

Reply:
Whoops, auto-formatting got it.  Fixed.